### DISCUSSION TITLE
Add some actions (clear query, copy path, commit)

### DISCUSSION THREAD

**Discussion by ikiet** [ibhagwan/fzf-lua#1403] [Ideas]
@ibhagwan 
First, thank you for this awesome plugin and your contributions to the community.

I switched to fzf-lua recently and I love it very much.
And when using it, I created some custom actions and thought someone would need it like me.
Ideas:
- Copy path action like Telescope (with absolute path and relative path)
- Clear current query & search (I am a fan of resume pickers so I need a way to clear the query instead of closing and reopening a fresh one)
- Git commit directly in git_status picker (like creating a new branch in git_branches)

So I want to know your opinion about these custom actions and I am willing to create PR for that.

**Comment by ibhagwan** [comment]
Ty for the kind words @ikiet.

Good ideas, although for something as simple as copying a path, I think it’s more beneficial to have it in the wiki so users can learn more how to customize?

> Git commit

This action should probably also come with an automatic header, although I recently added `action = { fn = …, header = … }` it’s not as automatic as I’d like it to be (I need to continue my refactoring of binds/actions).

> Clear current query & search

Clearing query already comes with fzf defaults set to `ctrl-u`, not sure if we need a specific action to clear both in one go?

**Comment by ikiet** [comment]
@ibhagwan

> Good ideas, although for something as simple as copying a path, I think it’s more beneficial to have it in the wiki so users can learn more how to customize?

Sounds good. I'll create a PR to add this to the wiki when I have time. (I'm still learning Lua and Neovim, so my speed is more like Flash 🦥)

> This action should probably also come with an automatic header, although I recently added action = { fn = …, header = … } it’s not as automatic as I’d like it to be (I need to continue my refactoring of binds/actions).

I’ll follow your refactoring and create git commit PR once I figure out how you want to refactor the code.

> Clearing query already comes with fzf defaults set to ctrl-u, not sure if we need a specific action to clear both in one go?

Oh, I didn’t realize that before, so I created a workaround to achieve it. Now, I’m using the default "unix-line-discard" action, and it works like a charm :))
